File: MORM.CSharp/Classes/Collection.cs
namespace MORM.Reppositorio.Classes
{
    public class Collection : IList<CollectionItem>
    {
        public Collection(Type collectionItemClass)
        {
            CollectionItemClass = collectionItemClass;
        }
        
        public Type CollectionItemClass { get; private set; }
        
        public CollectionItem AddItem()
        {
            var collectionItem = Activator.Create(CollectionItemClass);
            this.Add(collectionItem);
            return collectionItem;
        }
    }
}

File: MORM.Java/Classes/Collection.java
package MORM.Java.Classes;

public class Collection
{
    public Collection(Class collectionItemClass)
    {
        CollectionItemClass = collectionItemClass;
        Lista = new TList<CollectionItem>;
    }
    
    public Class CollectionItemClass;
    public TList<CollectionItem> Lista;
    
    public CollectionItem AddItem()
    {
        CollectionItem collectionItem;
        Lista.Add(collectionItem);
        return collectionItem;
    }
}